Zend/zend_gc.c

```diff
@@ -1465,14 +1465,17 @@ static int gc_remove_nested_data_from_buffer(zend_refcounted *ref, gc_root_buffe
 }
 
 static void zend_get_gc_buffer_release(void);
-static void zend_gc_root_tmpvars(void);
+static void zend_gc_check_root_tmpvars(void);
+static void zend_gc_remove_root_tmpvars(void);
 
 ZEND_API int zend_gc_collect_cycles(void)
 {
 	int total_count = 0;
 	bool should_rerun_gc = 0;
 	bool did_rerun_gc = 0;
 
+	zend_gc_remove_root_tmpvars();
```

Comment:
Can this be moved into the `if` below? Preferably after `if (GC_G(gc_active)) {` so that this doesn't run twice when GC isn't even active.

Reply:
It make sense to wrap this call with ``if (GC(active) && GC_G(num_roots))``.
Moving the call down is not a good idea because of ``rerun_gc`` label.